tests/integration/Core/Framework/DataAbstractionLayer/EntityExtensionTest.php

```diff
@@ -520,50 +521,74 @@ public function testICantAddScalarExtensions(): void
     {
         $this->registerDefinitionWithExtensions(ExtendableDefinition::class, FkFieldExtension::class);
-        $extension = static::getContainer()->get(ExtendableDefinition::class);
+        $definition = static::getContainer()->get(ExtendableDefinition::class);
 
-        static::assertInstanceOf(ExtendableDefinition::class, $extension);
-        static::assertTrue($extension->getFields()->has('test'));
+        static::assertInstanceOf(ExtendableDefinition::class, $definition);
+        static::assertTrue($definition->getFields()->has('test'));
     }
 
     public function testICanAddAssociationExtensions(): void
     {
         $this->registerDefinition(ExtendedDefinition::class);
         $this->registerDefinitionWithExtensions(ExtendableDefinition::class, AssociationExtension::class);
-        $extension = static::getContainer()->get(ExtendableDefinition::class);
+        $definition = static::getContainer()->get(ExtendableDefinition::class);
 
-        static::assertInstanceOf(ExtendableDefinition::class, $extension);
-        static::assertTrue($extension->getFields()->has('toOne'));
-        static::assertTrue($extension->getFields()->has('toMany'));
+        static::assertInstanceOf(ExtendableDefinition::class, $definition);
+        static::assertTrue($definition->getFields()->has('toOne'));
+        static::assertTrue($definition->getFields()->has('toMany'));
     }
 
     public function testICanAddReferenceVersionAsExtensionWithValidManyToOneAssociation(): void
     {
         $this->registerDefinition(ExtendedDefinition::class);
         $this->registerDefinitionWithExtensions(ExtendableDefinition::class, ReferenceVersionExtension::class);
-        $extension = static::getContainer()->get(ExtendableDefinition::class);
+        $definition = static::getContainer()->get(ExtendableDefinition::class);
 
-        static::assertInstanceOf(ExtendableDefinition::class, $extension);
-        static::assertTrue($extension->getFields()->has('toOne'));
-        static::assertTrue($extension->getFields()->has('extendedVersionId'));
+        static::assertInstanceOf(ExtendableDefinition::class, $definition);
+        static::assertTrue($definition->getFields()->has('toOne'));
+        static::assertTrue($definition->getFields()->has('extendedVersionId'));
+    }
+
+    public function testICanModifyFields(): void
+    {
+        $this->registerDefinitionWithExtensions(ExtendableDefinition::class, ModifyFieldsExtension::class);
+        $definition = static::getContainer()->get(ExtendableDefinition::class);
+
+        static::assertInstanceOf(ExtendableDefinition::class, $definition);
+        foreach ($definition->getFields() as $field) {
+            if ($field->getPropertyName() === 'apiAwareTest') {
+                static::assertFalse($field->is(ApiAware::class), 'Field ' . $field->getPropertyName() . ' should not have ApiAware flag');
+                break;
+            }
+        }
```

Comment:
we should add a check here, that the field `apiAwareTest` was really found. otherwise it is not 100% ensured, that `newField` was not added :slightly_smiling_face: